cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -453,21 +452,220 @@ private KafkaRebalanceStatus buildRebalanceStatusFromPreviousStatus(KafkaRebalan
+
+        for (Map.Entry<Integer, Map<String, Object>> loadBeforeEntry : loadBeforeMap.entrySet()) {
+            Map<String, Object> brokerBefore = loadBeforeEntry.getValue();
+            Map<String, Object> brokerAfter = loadAfterMap.get(loadBeforeEntry.getKey());
+
+            JsonObject brokerStats = new JsonObject();
+
+            for (CruiseControlLoadParameters intLoadParameter : CruiseControlLoadParameters.getIntegerParameters()) {
+
+                if (brokerBefore.containsKey(intLoadParameter.getKafkaRebalanceStatusKey()) &&
+                        brokerAfter.containsKey(intLoadParameter.getKafkaRebalanceStatusKey())) {
+
+                    int intBeforeStat = (int) brokerBefore.get(intLoadParameter.getKafkaRebalanceStatusKey());
+                    int intAfterStat = (int) brokerAfter.get(intLoadParameter.getKafkaRebalanceStatusKey());
+                    int intDiff = intAfterStat - intBeforeStat;
+
+
+                    JsonObject intStats = new JsonObject();
+                    intStats.put("before", intBeforeStat);
+                    intStats.put("after", intAfterStat);
+                    intStats.put("diff", intDiff);
+
+                    brokerStats.put(intLoadParameter.getKafkaRebalanceStatusKey(), intStats);
+                } else {
+                    log.warn("{} information was missing from the broker before/after load information",
+                            intLoadParameter.getKafkaRebalanceStatusKey());
+                }
+
+            }
+
+            for (CruiseControlLoadParameters doubleLoadParameter : CruiseControlLoadParameters.getDoubleParameters()) {
+
+                if (brokerBefore.containsKey(doubleLoadParameter.getKafkaRebalanceStatusKey()) &&
+                        brokerAfter.containsKey(doubleLoadParameter.getKafkaRebalanceStatusKey())) {
+
+                    double doubleBeforeStat = (double) brokerBefore.get(doubleLoadParameter.getKafkaRebalanceStatusKey());
+                    double doubleAfterStat = (double) brokerAfter.get(doubleLoadParameter.getKafkaRebalanceStatusKey());
+                    double doubleDiff = doubleAfterStat - doubleBeforeStat;
+
+                    JsonObject doubleStats = new JsonObject();
+                    doubleStats.put("before", doubleBeforeStat);
+                    doubleStats.put("after", doubleAfterStat);
+                    doubleStats.put("diff", doubleDiff);
+
+                    brokerStats.put(doubleLoadParameter.getKafkaRebalanceStatusKey(), doubleStats);
+                } else {
+                    log.warn("{} information was missing from the broker before/after load information",
+                            doubleLoadParameter.getKafkaRebalanceStatusKey());
+                }
+
+            }
+
+            brokersStats.put(String.valueOf(loadBeforeEntry.getKey()), brokerStats);
+        }
+
+        return brokersStats;
+    }
+
+    static class MapAndStatus<T, K> {
+
+        T loadMap;
+
```

Comment:
Not sure we need this space. Also maybe add a brief bit of javadoc/comment saying what this class is for?